src/components/Form.js

```diff
@@ -1,47 +1,85 @@
 import { useForm } from 'react-hook-form';
 import styled from 'styled-components';
 import { useImmer } from 'use-immer';
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import { DateRangeInput } from '@datepicker-react/styled';
 import { useReducer } from 'react';
 import { ThemeProvider } from 'styled-components';
+import MapboxGeocoder from '@mapbox/mapbox-gl-geocoder';
+import './Form.css';
+import mapboxgl from 'mapbox-gl';
 
 export default function Form({
   ButtonName,
   formName,
   initialState,
-  initialCount,
   destination,
   submit,
   preloadedValues,
 }) {
+  mapboxgl.accessToken =
+    'pk.eyJ1Ijoiai1tYXNzbWFubiIsImEiOiJjbDAyYzl5MHUwNW4yM2xxYmNrY3ViMmQ3In0.OFJuoFCLObuzC-2xN6gzZA';
+  const [locations, updateLocations] = useImmer(
+    destination?.locations ? destination.locations : []
+  );
+  const [destinationMapbox, setDestinationMapbox] = useState(
+    preloadedValues?.destination === undefined
+      ? ''
+      : preloadedValues.destination
+  );
+  const [locationMapbox, setLocationMapbox] = useState('');
+
+  useEffect(() => {
+    const geocoderDestination = new MapboxGeocoder({
+      accessToken: mapboxgl.accessToken,
+      types: 'country, region, place',
+      limit: 5,
+      placeholder: 'e.g. Lissabon',
+      minLength: 2,
+    });
+    geocoderDestination.on('result', e => {
+      setDestinationMapbox(e.result.text);
+    });
+    geocoderDestination.addTo('#geocoderdestination');
+
+    const geocoderLocations = new MapboxGeocoder({
+      accessToken: mapboxgl.accessToken,
+      types: 'region, place, poi',
+      limit: 5,
+      placeholder: 'Add a place you want to vist...',
+      minLength: 2,
+      clearOnBlur: true,
+    });
+    geocoderLocations.on('result', e => {
+      setLocationMapbox(e.result.text);
+    });
+
+    geocoderLocations.addTo('#geocoderlocation');
+  }, []);
+
   const {
-    register,
     handleSubmit,
     setError,
     formState: { errors },
   } = useForm({
     mode: 'all',
-    defaultValues: preloadedValues
-      ? preloadedValues
-      : { destination: '', locations: '' },
   });
   const [stateDate, dispatch] = useReducer(reducer, initialState);
-  const [counter, setCounter] = useState(initialCount);
-  const [locations, updateLocations] = useImmer(
-    destination?.locations ? destination.locations : []
-  );
-  const [disable, setDisable] = useState(true);
-  const onSubmit = data => {
-    submit(data, stateDate, locations);
+  const [destinationError, setDestinationError] = useState(false);
+
+  const onSubmit = () => {
+    submit(destinationMapbox, stateDate, locations);
   };
```

Comment:
Usually callback functions should be named with the prefix `on` and the function declaration without:
```suggestion
  const handleSubmit = () => {
    onCreateLocation(destinationMapbox, stateDate, locations);
  };
```
Maybe like this?